app/models/foreman_tasks/task.rb

```diff
@@ -158,6 +158,20 @@ def add_missing_task_groups(groups)
       (groups - task_groups).each { |group| task_groups << group }
     end
 
+    def sub_tasks_counts
+      result = %w(cancelled error pending success warning).zip([0].cycle).to_h
+      result.update sub_tasks.group(:result).count
+      sum = result.values.reduce(:+)
+      if respond_to?(:main_action) && main_action.respond_to?(:total_count)
+        result[:total] = main_action.total_count
+        key = state == 'stopped' ? 'cancelled' : 'pending'
```

Comment:
While we know what's going on here right now, I bet we will not be that clever few months later: please add an explanatory comment why we are doing this